### ISSUE TITLE
[Sormas2Sormas] Automatic Sync with every instance, when a shared dataset changed [3]

### ISSUE THREAD

**Issue by markusmann-vg** [hzi-braunschweig/SORMAS-Project#6019]
### Situation Description

Currently we are able to sync a shared set of data with another instance by triggering this action manually. This might lead to the situation that data, that was shared with another instance is not updated in the other instance but that would be necessary, even if it's send just for notification and without ownership.

### Feature Description

To be able to sync a shared dataset with the instances this dataset was shared we first have to know, with whom the dataset was shared. This means we have to know the whole chain. Example:

- Cluj is the owner of a case and sends it without ownership (for knowledge) to Palma.
- Palma shares the case with Braunschweig.
- Now Cluj has to know that the case was shared with Palma and Braunschweig.
- This means, **once a dataset was accepted**, the sender(s) and the "mother" of the case needs to be informed.

Now, as Cluj is the owner of the case and is able to edit it, we should automatically sync the case after it is saved. This means: 
The case is changed (e.g. add another symptom) by Cluj and saved
Then Palma should get the updated case information automatically (**the manual sync can be removed then, bulk sync action is no longer needed**) AND Braunschweig should get the information from Cluj about the updated case.

- [ ] remove the manual sync option from the UI

Exception:
We do not update the "deleted" information,  which means when Braunschweig deletes the case, then it's no longer visible for Braunschweig BUT the information, that was updated from Cluj is also updated in the deleted case in Braunschweig.
In Short: We sync everything but the "deleted" state.

 
### Additional Information

After the automatic sync also Cluj, Palma and Braunschweig see that the ownership changed and Berlin is now the "mother" of the case.

Prerequisite for this ticket https://github.com/hzi-braunschweig/SORMAS-Project/issues/6033

**Comment by markusmann-vg** [comment]
@kwa20 @JaquM Please check, if this fits your needs and add specification when needed

**Comment by vidi42** [comment]
Split the viewing of the shared chain in https://github.com/hzi-braunschweig/SORMAS-Project/issues/6033

> Exception:
> We do not update the "deleted" information, which means when Braunschweig deletes the case, then it's no longer visible for Braunschweig BUT the information, that was updated from Cluj is also updated in the deleted case in Braunschweig.
> In Short: We sync everything but the "deleted" state.

@markusmann-vg what do you mean exactly by this? Does this mean that once a case is soft deleted we need to still update it's information on the instance where it was deleted?

This part might be split in another ticket

**Comment by vidi42** [comment]
One important thing to mention is that at the moment only the instance which holds the ownership can share / delete a case.
